lib/active_merchant/billing/gateways/s5.rb

```diff
@@ -0,0 +1,236 @@
+require 'nokogiri'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class S5Gateway < Gateway
+      self.test_url = 'https://test.ctpe.io/payment/ctpe'
+      self.live_url = 'https://ctpe.io/payment/ctpe'
+
+      self.supported_countries = ['DK']
+      self.default_currency = 'EUR'
+      self.supported_cardtypes = [:visa, :master, :maestro]
+
+      self.homepage_url = 'http://www.s5.dk/'
+      self.display_name = 'S5'
+
+      SUPPORTED_TRANSACTIONS = {
+        'sale'      => 'CC.DB',
+        'authonly'  => 'CC.PA',
+        'capture'   => 'CC.CP',
+        'refund'    => 'CC.RF',
+        'void'      => 'CC.RV',
+      }
+
+      def initialize(options={})
+        requires!(options, :sender, :channel, :login, :password)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+        add_memo(post, options) if options[:memo]
+
+        commit('sale', post)
+      end
+
+      def refund(money, authorization, options={})
+        post = {}
+        post[:authorization] = authorization
+        add_invoice(post, money, options)
+        add_memo(post, options) if options[:memo]
```

Comment:
I'm thinking you can put the conditional of `if options[:memo]` in the `add_memo` method to simplify.  Then you can remove the other few times the conditional is here.